fix filetree fix function naming relocate tests some class renaming

Rorzakh-tix opened the pull request.

Rorzakh-tix committed:
fix filetree fix function naming relocate tests some class renaming

mikhaillazko commented on src/library/database/models/base_model.py:
let's remove dir models and move file to parent dir - database

Rorzakh-tix replied:
done

mikhaillazko commented on src/library/database/database_config.py:
```diff
@@ -25,12 +23,12 @@ def assemble_db_connection(cls, v: Optional[str], info: ValidationInfo) -> str:
 
     model_config = SettingsConfigDict(
-        env_file='.env',
+        env_file='../../.env',
```
bad practice to use relative path, probably you want to use something like: `.env`

Rorzakh-tix replied:
done

mikhaillazko commented on src/library/minio_aws/minio_config.py:
minio is just one of the implementation s3 protocol and don't used by AWS, let's rename folder from `minio_aws` to `s3` or `file_storage`

Rorzakh-tix replied:
renamed to s3

mikhaillazko commented on src/scrapper/mortgage.py:
```diff
@@ -13,7 +13,24 @@ class MortgageInputData(BaseModel):
+        browser = await p.chromium.launch(headless=False, slow_mo=300)
+        context = await browser.new_context(ignore_https_errors=True,
+                                            user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 "
+                                                       "(KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36")
```
**nit**: ugly style, as option you can extract this long string to constant

Rorzakh-tix replied:
extracted to constant

mikhaillazko commented on src/scrapper/mortgage.py:
```diff
@@ -13,7 +13,24 @@ class MortgageInputData(BaseModel):
+        page = await _fill_mortgage_calc_fields(page, mortgage_data)
+        paymentsDictList = await _get_table_from_page_content(page)
+        await _get_result_page_pdf(page)
+        await context.close()
```
most of the time when you need to call close method i would guess it should support context manager `with` out of the box

Rorzakh-tix replied:
rewrote with context manager

mikhaillazko commented on src/s3/__init__.py:
looks like can be removed

Rorzakh-tix replied:
done

mikhaillazko commented on src/scrapper/mortgage.py:
the better name is `mortgage_parser` or `mortgage_scraper`

Rorzakh-tix replied:
renamed to mortgage_scrapper

mikhaillazko commented on tests/__init__.py:
can be removed

Rorzakh-tix replied:
done

mikhaillazko reviewed (commented):
i  was expect that schemas will be removed but i still see it in apps

Rorzakh-tix committed:
pull request fixes #1

Rorzakh-tix committed:
pull request fixes #2 delete __init__.py from src

Rorzakh-tix commented:
> i was expect that schemas will be removed but i still see it in apps

removed mortgage_schemes

mikhaillazko commented on src/main.py:
```diff
@@ -24,7 +24,7 @@ async def lifespan(app: FastAPI):
 app.include_router(fastapi_users_ep.get_verify_router(UserRead), prefix="/auth", tags=["auth"])
 app.include_router(fastapi_users_ep.get_users_router(UserRead, UserUpdate), prefix="/auth", tags=["auth"])
-app.include_router(router=mortgage_router,prefix="/scrap", tags=["scraping"])
+app.include_router(router=mortgage_router, prefix="/scrap", tags=["scraping"])
```
nit: `/scrap` -> `/scrapper`

Rorzakh-tix replied:
changed the prefix /scrap to /scrapper in mortgage_router

mikhaillazko commented on src/scrapper/mortgage_endpoints.py:
```diff
@@ -0,0 +1,30 @@
+)
+
+
+@mortgage_router.post("/")
```
please add separate route `/scrap` or `/parse`

Rorzakh-tix replied:
added separate route "/scrap" instead of "/" for mortgage_router

mikhaillazko commented on src/conftests.py:
```diff
@@ -21,8 +21,8 @@
 
-@pytest.fixture(scope="function")
-async def prepare_database():
+@pytest.fixture(scope="session")
```
it should have scope=function if we name it as session in DB terms

mikhaillazko replied:
looks like you want to have two separate fixture one for setup database when tests running and clean up when all test complete and another fixture to work with database in each test case 
```
def setup_database(): ... - have session scope
def session(setup_database): .... - have function scope
```

Rorzakh-tix replied:
Moved "setup_database()" fixture into scope=session. Add "session" fixture with scope=function. Removed unnecessary functions and constants. Removed print function

mikhaillazko commented:
also would better provide more human readable name for pull request and fill description with detail bullet list what was done

Rorzakh-tix committed:
pull request fixes #3
changed the prefix in mortgage_router
added separate route "/scrap" instead of "/" for mortgage_router
Moved "setup_database()" fixture into scope=session
Add "session" fixture with scope=function
Removed unnecessary functions and constants
Removed print functions
